flask/app.py

```diff
@@ -2063,19 +2064,26 @@ def make_response(self, rv):
                 status = headers = None
             elif isinstance(rv, dict):
                 rv = jsonify(rv)
-            else:
+            elif isinstance(rv, BaseResponse) or hasattr(rv, "__call__"):
```

Comment:
Better to use `callable()` here rather than hasattr(..., '__call__').

(It would actually be `hasattr(obj.__class__, '__call__')` to test if something is callable.)

Reply:
Updated to use `callable()`. I was trying to support Python 3.0/3.1 😜

Reply:
(@lordmauve, agreed this should use `callable`, but your `hasattr(obj.__class__, '__call__')` looks like it returns true no matter what `obj` is, no?)

Reply:
Pretty sure those are EOL :P (and in any case, they are unsupported by the pallets projects)

Reply:
@jab You are right. Just trying to get to the point that dunder methods are looked up on the type of the thing, not the thing itself.